mlir-sycl/lib/Dialect/SYCL/IR/SYCLHostOps.cpp

```diff
@@ -10,3 +10,22 @@ LogicalResult SYCLHostConstructorOp::verify() {
            << type;
   return success();
 }
+
+void SYCLHostConstructorOp::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  // NOTE: This definition assumes only the first (`this`) argument is written
+  // to and the remaining arguments, if they are memory arguments, are both read
+  // and written. The latter is a conservative assumption about the behavior of
+  // the constructors and could be changed to enable better analysis.
+  auto *defaultResource = SideEffects::DefaultResource::get();
+  // The `this` argument will always be written to
+  effects.emplace_back(MemoryEffects::Write::get(), getDst(), defaultResource);
```

Comment:
The `this` argument is also allocated.

Reply:
No, in `sycl.host.constructor` it's not allocated, but passed as the first argument, e.g., coming from an `llvm.alloca`.